### ISSUE TITLE
[0781 PS] [4.2] Medical records choice page

### ISSUE THREAD

**Issue by lisacapaccioli** [department-of-veterans-affairs/va.gov-team#98813]
**Precondition**: [4.1] Evidence orientation (content only page)

### Acceptance criteria

As a user, I need an option to identify what type of medical records I want to add to my claim. 

**First time to the page:** 

- If the user selects “continue” from [4.1] Evidence orientation direct them to the [4.2] Medical records choice page with all options disabled. 
   -  Multiple select options
      1.  Yes - user needs to request copies ("VA") 
      2. Yes - user needs VA to request copies ("Non-VA") 
      3. Yes - user wants to upload copies ("Upload non-VA") 
      4. No - user doesn't need a review ("No") 

- Selecting the back button returns the user to the [4.1] Evidence orientation page
- Selecting continue requires the user to have selected an option.
  - There are two different error messages:  
    - If the user doesn't select an option and selects to continue display an error message and don't let them continue
    - If the user selects a "yes" option and the "no" option,  display an error message and don't let them continue
    - A11Y: on error direct the focus to the error message
  - If the user selecting an option and then continue follow the follow flows

**- Flow from this page is conditional**
  - If the user selects the Yes "VA" option (#a above) direct the user to the [4.3] VA treatment question page
  - If the user selects the "non-VA" option (#b above) direct the user to the [4.5] VA treatment summary page
  - If the user selects the "upload" option (#c above) direct the user to the [4.8] Add non-VA treatment add/edit
  - If the user selects multiple options, go in this order:  
     - [4.3] VA treatment question page
     - [4.5] VA treatment summary page
     - [4.8] Add non-VA treatment add/edit

**Returning to the page:** 
[4.2] Medical records choice page with previously selected options enabled 

Once the selections are made there is more for tracking the data (next ticket). 
4142 — PMRs (Private medical records) only
526 — VAMC (VA Medical treatment location) (for treatment not related to traumatic events) only
0781 — VAMC AND PMRs relating to treatment for traumatic events

### Questions
The 0781 asks for treatment locations. it says if it is a VA treatment center, it can go in the section on the 0781 only. But if it is an external provider, that a form 4142 must also be filled out. We (potentially) already fill out a form 4142 for the overall 526, if they gave treatment locations for any other stuff they are claiming that is not 0781 related.

What we think we know:

- If the user provides only VA providers, we just put them in the space provided on the 0781

What we don't think we know: 

- If the user provides non-VA treatment locations, do we fill out a 4142 for them like we do for the broader 526
- If we do should we do 2x 4142s (one for broader 526, one for 0781 treatment places only)
- or should we combine both onto a single 4142
- Do non-VA locations for 0781 still get written on the 0781, or do they only get written on the 4142

_We’ll anchor whether to write the medical record request to the 4142, 526 or 0781 based on whether or not treatment in the facility relates to their traumatic events (see question 13A).
If yes, then write to 0781, if no, then write to 4142 or 526 respectively (PMRs get written to 4142, VAMC or DOD record requests get written to the 526)._

**Comment by lisacapaccioli** [comment]
FYI: 4.1 Evidence orientation page exists today in production, below is that is production today: 
![Image](https://github.com/user-attachments/assets/45efa2f8-f515-4907-8b5c-dc0abee3c6b1)

**Comment by freeheeling** [comment]
@lisacapaccioli 
1. Perhaps the page has changed, but I don't see the following option on this page: 
> Yes - "Upload"

2. The following seems to contradict the user's preference. Shouldn't they be routed to the next section, Supporting documents?
> If the user selects "no" option (#d above) direct the user to the [4.8] Upload

And just a note that https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=11660-71080&t=HOFjG0bxY1d9Dj9l-4 does not depict any branching based on a user's selection on this page.

**Comment by lisacapaccioli** [comment]
Good catch. Will need to do refinement w/Ruben because it looks like it's been changed since I worked on this. There used to be 4 options, and now there's 3. So my guess is that the "no" would take them to the upload (which it looks like there are two now, so it would be last one - page 4.9).

**Comment by lisacapaccioli** [comment]
The options have been changed in the designs as well as more pages to the PMR flow. I updated this ticket based on the most current changes.
